core/src/main/java/com/amplifyframework/api/graphql/GraphQLResponse.java

```diff
@@ -83,6 +94,23 @@ public boolean hasData() {
         return data != null;
     }
 
+    /**
+     * Gets the token for fetching more results, if there are any.
+     * @return nextToken
+     */
+    @Nullable
+    public String getNextToken() {
+        return nextToken;
+    }
+
+    /**
+     * Checks if a nextToken exists, indicating more results can be fetched.
+     * @return true if nextToken is not null, false otherwise
+     */
+    public boolean hasNextToken() {
+        return nextToken != null;
+    }
+
     @Override
     public boolean equals(Object thatObject) {
```

Comment:
Don't forget to update `equals()`, `hashCode()` `toString()`, as well, with the new field.